trace_test.go

```diff
@@ -21,6 +21,7 @@ import (
 	"context"
 	"errors"
 	"fmt"
+	"github.com/go-test/deep"
```

Comment:
Do this need this extra dependency? There must be something in the `require` package we already use for deep comparisons.